### PR TITLE
Code refactor on marketing components

### PR DESCRIPTION
**Pull request by ecgan**
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:

This PR contains code refactor on marketing components:

- `CardHeaderTitle` component is introduced so that it can be used outside of `CollapsibleCard` component.
- `CenteredSpinner` component is introduced so that we can easily use it anywhere without having to repeat CSS for it.
- There were two `Plugin` types. They are renamed to `InstalledPlugin` and `RecommendedPlugin` to make them clearer and avoid confusion.
- `PluginCardBody` is modified to accept `className` props, and its `description` prop type is changed from `string` type to `ReactNode` type to make it more flexible and to be used in later PRs.

The changes introduced here would be used in subsequent PRs for new features in Multichannel Marketing page. It should not break existing functionality in the page.

### How to test the changes in this Pull Request:

Pre-condition: 

- Enable WooCommerce Multichannel Marketing experience in WooCommerce Settings > Advanced > Features (`/wp-admin/admin.php?page=wc-settings&tab=advanced&section=features`).

Test instructions:

1. Go to Marketing page at `/wp-admin/admin.php?page=wc-admin&path=%2Fmarketing`.
2. Click around in the page and things should be working as expected, e.g.:
    - One-click installation should work.
    - Clicking on tabs should work.
    - Clicking on buttons should work as expected.

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?
-   [x] Have you created a changelog file for each project being changed, ie `pnpm --filter=<project> changelog add`?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

File: plugins/woocommerce-admin/client/marketing/components/CardHeaderTitle/CardHeaderTitle.scss [hunk 1]
```diff
@@ -0,0 +1,6 @@
+.woocommerce-marketing-card-header-title {
+	font-size: 20px;
+	font-weight: 400;
+	line-height: 28px;
+	letter-spacing: 0;
+}
```
Comment (puntope) [inline plugins/woocommerce-admin/client/marketing/components/CardHeaderTitle/CardHeaderTitle.scss]: 💅  What about using some variables here?

**Review by puntope — DISMISSED** [review]
✅  LGTM

Thanks for the refactoring. I left some comment regarding variables. But not a blocker-

File: plugins/woocommerce-admin/client/marketing/components/CardHeaderTitle/CardHeaderTitle.scss (hunk 1 above)
Comment (ecgan) [reply plugins/woocommerce-admin/client/marketing/components/CardHeaderTitle/CardHeaderTitle.scss]: @puntope , thanks, I simplified the code in [`c2aad26` (#36540)](https://github.com/woocommerce/woocommerce/pull/36540/commits/c2aad264734ebe858a44cef9c6597037b6d77b80):

- Use `@include font-size ( 20 )` to generate relative size in rem.
- Use unitless number value for line-height; this is the preferred way according to [MDN](https://developer.mozilla.org/en-US/docs/Web/CSS/line-height).
- `font-weight` and `letter-spacing` are not needed.

The UI looks the same before and after this change.

**Comment by ecgan** [comment]
@puntope , thanks for the review. I made a code change to address your comment, and your approving review was dismissed. Could you do another round of review please?

**Review by puntope — APPROVED** [review]
Howdy @ecgan 

Looking good. I re-run the failing tests (sometimes they fail rendomly)

**Pull request merged**